### PR TITLE
Added LTC4150 Demo

### LINKED ISSUE #982: Add LTC4150 Demo Project
I can do this!

### PR DESCRIPTION
**Pull request by evanugarte**
Now the user can quickly get set up working with the coulomb counter. The demo includes documentation for constructing, initializing and reading charge from the LTC4150.

Resolves #982

### CODE REVIEW

File: demos/sjtwo/ltc4150/source/main.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  constexpr float kResolution = 0.001f;
+  LOG_INFO("Ltc4150 application starting...");
+  // Creating GPIO on pin 2.0
+  sjsu::lpc40xx::Gpio tick_pin(2, 0);
+  // Creating GPIO on pin 2.1
+  sjsu::lpc40xx::Gpio polarity_pin(2, 1);
+  // Set the sense resistor resistance to be 50 milliohms. This will provide a
+  // 50 millivolt drop at maximum charge or discharge to invoke an interrupt on
+  // the tick_pin.
+  units::impedance::ohm_t resistance = 0.00050_Ohm;
```
Comment (kammce) [inline demos/sjtwo/ltc4150/source/main.cpp]:
```suggestion
  units::impedance::milliohm_t resistance = 50_mOhm;
```
Use a more readable unit.

File: demos/sjtwo/ltc4150/source/main.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+  constexpr float kResolution = 0.001f;
+  LOG_INFO("Ltc4150 application starting...");
+  // Creating GPIO on pin 2.0
+  sjsu::lpc40xx::Gpio tick_pin(2, 0);
+  // Creating GPIO on pin 2.1
+  sjsu::lpc40xx::Gpio polarity_pin(2, 1);
+  // Set the sense resistor resistance to be 50 milliohms. This will provide a
+  // 50 millivolt drop at maximum charge or discharge to invoke an interrupt on
+  // the tick_pin.
```
Comment (kammce) [inline demos/sjtwo/ltc4150/source/main.cpp]: Most of this does not make sense with regards to how the chip works. Remove everything except for this line "// Set the sense resistor resistance to be 50 milliohms."

File: demos/sjtwo/ltc4150/source/main.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  units::impedance::ohm_t resistance = 0.00050_Ohm;
+  // Initialize the LTC4150 with the two GPIO pins and resistance.
+  sjsu::Ltc4150 counter(tick_pin, polarity_pin, resistance);
+  // Initialize the LTC4150 class, setting two GPIO pins to be outputs and
```
Comment (kammce) [inline demos/sjtwo/ltc4150/source/main.cpp]: Add a newline here to separate between construction of the object and using it.

File: demos/sjtwo/ltc4150/source/main.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  // attaching interrupts.
+  counter.Initialize();
+  // Set the previous charge to be 0 mAh.
+  units::charge::milliampere_hour_t previous_charge = 0_mAh;
+
+  while (true)
+  {
+    // Grab the calculated charge from the counter, based on how many ticks it
+    // received and the polarity.
+    units::charge::milliampere_hour_t current_charge = counter.GetCharge();
```
Comment (kammce) [inline demos/sjtwo/ltc4150/source/main.cpp]:
```suggestion
    units::charge::milliampere_hour_t recent_charge = counter.GetCharge();
```
The name "current" here is kinda confusing as we are dealing with electrical current and temporal current.

File: demos/sjtwo/ltc4150/project.mk [hunk 5]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (kammce) [inline demos/sjtwo/ltc4150/project.mk]: Remove extra new line here.

File: demos/sjtwo/ltc4150/source/main.cpp (hunk 2 above)
Comment (evanugarte) [reply demos/sjtwo/ltc4150/source/main.cpp]: Done.

File: demos/sjtwo/ltc4150/project.mk (hunk 5 above)
Comment (evanugarte) [reply demos/sjtwo/ltc4150/project.mk]: Done.

File: demos/sjtwo/ltc4150/source/main.cpp (hunk 3 above)
Comment (evanugarte) [reply demos/sjtwo/ltc4150/source/main.cpp]: Done.

**Pull request merged**